crates/common/mqtt_channel/src/config.rs

```diff
@@ -164,7 +166,7 @@ impl Default for Config {
             subscriptions: TopicFilter::empty(),
             clean_session: false,
             queue_capacity: 1024,
-            max_packet_size: 1024 * 1024,
+            max_packet_size: MAX_PACKET_SIZE,
```

Comment:
Looking closer to root cause of an infinite loop caused by a message that is too large, it appears that the issue is not rooted in the MQTT protocol per se, but is due to a bad combination of rumqttc and thin-edge.

- when a message that is too large is received, rumqttc raises a `PayloadSizeLimitExceeded` error
- when this error is received thin-edge do nothing beyond logging and adding some delay (which is useless) 
- rumqttc also cuts the connection and then reconnects.
- the message is then resent by the broker that is still expecting an acknowledgement.

Unfortunately, the rumqttc API provides no means to skip that poison pill (the error structs does not contain the message pkid).
 
=> Removing the limit as done here fixes the issue. However, we can consider to add our own check with no limit on what is received from rumqttc but with a filter implemented by `mqtt_channel` to prune messages that are too large. Without this limit a rogue client might overwhelmed the mapper or the agent.

Reply:
Resolved a4a6af90e047bae0e82be752512648b90ec71583